panel/js/katalon/newformatters/protractorts.js

```diff
@@ -0,0 +1,177 @@
+
+
+newFormatters.protractorts = function(name, commands) {
+    var content = newprotractorts(name).formatter(commands);
+    return {
+      content: content,
+      extension: 'ts',
+      mimetype: 'text/javascript'
+    }
+  }
+
+  const newprotractorts = function (scriptName){
+    var _scriptName = scriptName  || "";
+    const locatorType = {
+
+        xpath: (target) => {
+            return `by.xpath("${target.replace(/\"/g, "\'")}")`
+        },
+        css: (target) => {
+            return `by.css("${target.replace(/\"/g, "\'")}")`
+        },
+    
+        id: (target) => {
+            return `by.id("${target.replace(/\"/g, "\'")}")`
+        },
+    
+        link: (target) => {
+            return `by.linkText("${target.replace(/\"/g, "\'")}")`
+        },
+    
+        name: (target) => {
+            return `by.name("${target.replace(/\"/g, "\'")}")`
+        },
+    
+        tag_name: (target) => {
+            return `by.tagName("${target.replace(/\"/g, "\'")}")`
+        }
+    }
+
+    // https://seleniumhq.github.io/selenium/docs/api/javascript/module/selenium-webdriver/index_exports_Key.html
+    const specialKeyMap = {
+        '\${KEY_LEFT}': 'Key.ARROW_LEFT',
+        '\${KEY_UP}': 'Key.ARROW_UP',
+        '\${KEY_RIGHT}': 'Key.RIGHT',
+        '\${KEY_DOWN}': 'Key.DOWN',
+        '\${KEY_PAGE_UP}': 'Key.PAGE_UP',
+        '\${KEY_PAGE_DOWN}': 'Key.PAGE_DOWN',
+        '\${KEY_BACKSPACE}': 'Key.BACK_SPACE',
+        '\${KEY_DEL}': 'Key.DELETE',
+        '\${KEY_DELETE}': 'Key.DELETE',
+        '\${KEY_ENTER}': 'Key.ENTER',
+        '\${KEY_TAB}': 'Key.TAB',
+        '\${KEY_HOME}': 'Key.HOME'
+    }
+    
+    // protracto api
+    // https://www.protractortest.org/#/api
+    // katalon
+    // https://docs.katalon.com/katalon-recorder/docs/selenese-selenium-ide-commands-reference.html
+    const seleneseCommands = {
+        "open": "await browser.get('_TARGET_');",
+        "click": "await element(_BY_LOCATOR_).click();",
+        "clickAndWait": 
+            "const el__STEP_ = element(_BY_LOCATOR_);\n" +
+            "\t\tawait browser.wait(EC.elementToBeClickable(el__STEP_));\n" + 
+            "\t\tawait await el__STEP_.click();",
```

Comment:
It's a bug. You should remove one `await` on this line.

Reply:
Thank you @doggy8088.

It has been fixed by @minhthanh3145.